source/slang/slang-ir-legalize-types.cpp

```diff
@@ -2344,6 +2344,10 @@ static LegalVal legalizeCoopMatMapElementIFunc(
 
 static LegalVal legalizeInst(IRTypeLegalizationContext* context, IRInst* inst)
 {
+    LegalVal legalVal;
+    if (context->mapValToLegalVal.tryGetValue(inst, legalVal))
+        return legalVal;
+
```

Comment:
I'm kind of shocked that not checking the cache here hadn't caused huge problems before now.